pkgs/by-name/am/amnezia-vpn/package.nix

```
@@ -0,0 +1,173 @@
+      substituteInPlace client/protocols/openvpnovercloakprotocol.cpp \
+        --replace-fail 'return Utils::executable(QString("/ck-client"), true);' 'return Utils::executable(QString("${cloak-pt}/bin/ck-client"), false);'
+      # fix openvpn over ss protocol
+      substituteInPlace client/protocols/shadowsocksvpnprotocol.cpp \
+        --replace-fail 'return Utils::executable(QString("/ss-local"), true);' 'return Utils::executable(QString("${shadowsocks-rust}/bin/sslocal"), false);'
+      # fix update-resolv-conf.sh path
+      substituteInPlace client/configurators/openvpn_configurator.cpp \
+        --replace-fail ".arg(qApp->applicationDirPath());" ".arg(\"$out/local/bin\");"
+      # fix icon path
+      substituteInPlace client/ui/qautostart.cpp \
+        --replace-fail "/usr/share/pixmaps/AmneziaVPN.png" "$out/share/pixmaps/AmneziaVPN.png"
+      # fix desktop item
+      substituteInPlace deploy/installer/config/AmneziaVPN.desktop.in \
+        --replace-fail "#!/usr/bin/env xdg-open" "#!${xdg-utils}/bin/xdg-open" \
+        --replace-fail "/usr/share/pixmaps/AmneziaVPN.png" "$out/share/pixmaps/AmneziaVPN.png"
+    ''
+    + (
+      if stdenv.hostPlatform.isAarch64 && stdenv.hostPlatform.isLinux then
+        ''
+          substituteInPlace client/cmake/3rdparty.cmake \
+            --replace-fail 'set(LIBSSH_LIB_PATH "''${LIBSSH_ROOT_DIR}/linux/x86_64/libssh.a")' 'set(LIBSSH_LIB_PATH "${libssh}/lib/libssh.so")' \
+            --replace-fail 'set(ZLIB_LIB_PATH "''${LIBSSH_ROOT_DIR}/linux/x86_64/libz.a")' 'set(ZLIB_LIB_PATH "${zlib}/lib/libz.so")' \
+            --replace-fail 'set(OPENSSL_LIB_SSL_PATH "''${OPENSSL_ROOT_DIR}/linux/x86_64/libssl.a")' 'set(OPENSSL_LIB_SSL_PATH "''${OPENSSL_ROOT_DIR}/linux/arm64/libssl.a")' \
+            --replace-fail 'set(OPENSSL_LIB_CRYPTO_PATH "''${OPENSSL_ROOT_DIR}/linux/x86_64/libcrypto.a")' 'set(OPENSSL_LIB_CRYPTO_PATH "''${OPENSSL_ROOT_DIR}/linux/arm64/libcrypto.a")'
+        ''
+      else
+        ""
+    );
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    qt6.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    libsecret
+    qt6.qtbase
+    qt6.qttools
+    kdePackages.qtremoteobjects
+    kdePackages.qtsvg
+    kdePackages.qt5compat
+  ];
+
+  qtWrapperArgs = [
+    ''--prefix PATH : ${
+      lib.makeBinPath [
+        busybox
+        sudo
+      ]
+    }''
+  ];
+
+  postInstall = ''
+    # install AmneziaVPN and AmneziaVPN-service
+    mkdir -p $out/bin
+    cp client/AmneziaVPN $out/bin/
+    cp service/server/AmneziaVPN-service $out/bin/
+    # install update-resolv-conf.sh
+    mkdir -p $out/local/bin
+    cp ../deploy/data/linux/client/bin/update-resolv-conf.sh $out/local/bin/
+    # install .desktop item
+    mkdir -p $out/share/applications
+    cp ../AppDir/AmneziaVPN.desktop $out/share/applications/
+    # install icon
+    mkdir -p $out/share/pixmaps/
+    cp ../deploy/data/linux/AmneziaVPN.png $out/share/pixmaps/
```

```suggestion
    mkdir -p $out/bin $out/local/bin $out/share/applications $out/share/pixmaps/
    cp client/AmneziaVPN service/server/AmneziaVPN-service $out/bin/
    cp ../deploy/data/linux/client/bin/update-resolv-conf.sh $out/local/bin/
    cp ../AppDir/AmneziaVPN.desktop $out/share/applications/
    cp ../deploy/data/linux/AmneziaVPN.png $out/share/pixmaps/
```
$out/locan is a pretty none standard location for us

---

Is this ok? I don't want `update-resolv-conf.sh` to be in `bin/` because this script is not intended to be used outside the package